classlib/src/main/java/org/teavm/classlib/java/util/TDeque.java

```diff
@@ -54,4 +43,244 @@ public interface TDeque<E> extends TQueue<E> {
     E pop();
 
     TIterator<E> descendingIterator();
+
+    @Override
+    default TDeque<E> reversed() {
+        return new ReversedDeque<>(this);
+    }
+
+    class ReversedDeque<E> implements TDeque<E> {
```

Comment:
I think this class should be moved to top level with package-private visibility. Rationale: declaring class in an interface makes this class public, which can in turn have consequences with reflection.